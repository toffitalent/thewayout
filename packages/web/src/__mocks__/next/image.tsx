import * as React from 'react';

// eslint-disable-next-line react/jsx-no-useless-fragment
const mock = (props: { children: React.ReactElement }): React.ReactElement => <>{props.children}</>;

export default mock;
