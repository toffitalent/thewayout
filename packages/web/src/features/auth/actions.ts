import { CreateUserRequest } from '@two/shared';
import { API } from '@app/api';
import { createApiThunk } from '@app/utils';

export const fetchUser = createApiThunk('auth/fetchUser', API.user.retrieve);
export const forgotPassword = createApiThunk('auth/forgotPassword', API.user.forgotPassword);

export const login = createApiThunk(
  'auth/login',
  async (credentials: Parameters<typeof API.login>[0], { dispatch }) => {
    await API.login(credentials);
    await dispatch(fetchUser());
  },
);

export const logout = createApiThunk('auth/logout', API.logout);

export const resetPassword = createApiThunk(
  'auth/resetPassword',
  async (
    { username, password, token }: { username: string; password: string; token: string },
    { dispatch },
  ) => {
    await API.user.resetPassword(password, token);
    return dispatch(login({ username, password }));
  },
);

export const signUp = createApiThunk(
  'auth/register',
  async (payload: CreateUserRequest, { dispatch }) => {
    await API.user.create(payload);
    return dispatch(login({ username: payload.email, password: payload.password }));
  },
);

export const updateUser = createApiThunk('auth/updateUser', API.user.update);
