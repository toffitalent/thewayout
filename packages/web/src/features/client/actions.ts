import { ListRspClientRequest } from '@two/shared';
import { API } from '@app/api';
import type { State } from '@app/store';
import { createApiThunk } from '@app/utils';

export const createClient = createApiThunk('client/create', API.clientProfile.create);

export const updateClient = createApiThunk('client/update', API.clientProfile.update);

export const listAllJobs = createApiThunk(
  'client/jobs',
  ({ forceReload }: { forceReload?: boolean }, { getState }) => {
    const { ids } = (getState() as State).clientJobs.jobsList;
    const lastId = [...ids].pop() as string;
    return API.job.list({
      ...(!forceReload && { before: lastId }),
    });
  },
);

export const retrieveJob = createApiThunk(
  'client/job',
  ({ jobId, since }: { jobId: string; since?: Date }) =>
    API.job.retrieve(jobId, {
      ...(since ? { since } : {}),
    }),
);

export const applyJob = createApiThunk(
  'client/jobApply',
  ({ jobId, questions }: { jobId: string; questions?: { [key: string]: string }[] }) =>
    API.job.apply(jobId, {
      ...(questions ? { questions } : {}),
    }),
);

export const listAppliedJobs = createApiThunk(
  'client/applications',
  ({ forceReload, userId }: { forceReload?: boolean; userId: string }, { getState }) => {
    const { ids } = (getState() as State).clientJobs.appliedJobs;
    const lastId = [...ids].pop() as string;
    return API.clientProfile.listJobs(userId, {
      ...(!forceReload && { before: lastId }),
    });
  },
);

export const listSuggestedJobs = createApiThunk('client/suggestedJobs', (clientId: string) =>
  API.clientProfile.listSuggestedJobs(clientId),
);

export const listRsp = createApiThunk(
  'rsp/list',
  ({ userId, ...rest }: ListRspClientRequest & { userId: string }) =>
    API.clientProfile.listRsp({ userId, ...rest }),
);

export const retrieveCaseManager = createApiThunk(
  'client/caseManager',
  API.clientProfile.getCaseManager,
);
