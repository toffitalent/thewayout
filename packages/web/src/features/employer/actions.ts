import { Job, JobApplicationStatus } from '@two/shared';
import { API } from '@app/api';
import type { State } from '@app/store';
import { createApiThunk } from '@app/utils';

export const createEmployer = createApiThunk('employer/create', API.employer.create);

export const updateEmployer = createApiThunk('employer/update', API.employer.update);

export const createJob = createApiThunk('employer/job/create', API.job.create);

export const listJobs = createApiThunk(
  'employer/jobs',
  ({ forceReload, userId }: { forceReload?: boolean; userId: string }, { getState }) => {
    const { ids } = (getState() as State).employerJobs.jobs;
    const lastId = [...ids].pop() as string;
    return API.employer.listJobs(userId, {
      ...(!forceReload && { before: lastId }),
    });
  },
);

export const updateListJobs = createApiThunk(
  'employer/jobs/listUpdated',
  ({ userId }: { userId: string }, { getState }) => {
    const { listUpdatedAt, ids } = (getState() as State).employerJobs.jobs;
    const firstId = ids[0] as string;
    const lastId = [...ids].pop() as string;
    return API.employer.listJobs(userId, {
      since: listUpdatedAt,
      after: firstId,
      before: lastId,
    });
  },
);

export const retrieveJob = createApiThunk(
  'employer/job',
  ({ jobId, since }: { jobId: string; since?: Date }) =>
    API.job.retrieve(jobId, {
      ...(since ? { since } : {}),
    }),
);

export const listApplications = createApiThunk(
  'employer/listApplications',
  ({ employerId, jobId }: { employerId: string; jobId: string }, { getState }) => {
    const { entities } = (getState() as State).employerJobs.jobsApplications;
    const page = entities[jobId]?.page;
    const isListEmpty = !entities[jobId]?.applications.length;

    return API.employer.listApplications(employerId, jobId, {
      page: isListEmpty ? 0 : page! + 1,
    });
  },
);

export const getClient = createApiThunk(
  'employer/getClient',
  ({ employerId, jobId, clientId }: { employerId: string; jobId: string; clientId: string }) =>
    API.employer.getClient(employerId, jobId, clientId, {}),
);

export const updateJob = createApiThunk(
  'employer/updateJob',
  ({ jobId, patch }: { jobId: string; patch: Partial<Job> }) => API.job.update(jobId, patch),
);

export const interviewClient = createApiThunk(
  'employer/interviewClient',
  ({
    employerId,
    jobId,
    applicationId,
  }: {
    employerId: string;
    jobId: string;
    applicationId: string;
  }) =>
    API.employer.changeApplicationStatus(
      employerId,
      jobId,
      applicationId,
      JobApplicationStatus.interview,
    ),
);

export const hireClient = createApiThunk(
  'employer/hireClient',
  ({
    employerId,
    jobId,
    applicationId,
  }: {
    employerId: string;
    jobId: string;
    applicationId: string;
  }) =>
    API.employer.changeApplicationStatus(
      employerId,
      jobId,
      applicationId,
      JobApplicationStatus.hired,
    ),
);

export const rejectClient = createApiThunk(
  'employer/rejectClient',
  ({
    employerId,
    jobId,
    applicationId,
  }: {
    employerId: string;
    jobId: string;
    applicationId: string;
  }) =>
    API.employer.changeApplicationStatus(
      employerId,
      jobId,
      applicationId,
      JobApplicationStatus.rejected,
    ),
);

export const notAFitClient = createApiThunk(
  'employer/notAFitClient',
  ({
    employerId,
    jobId,
    applicationId,
  }: {
    employerId: string;
    jobId: string;
    applicationId: string;
  }) =>
    API.employer.changeApplicationStatus(
      employerId,
      jobId,
      applicationId,
      JobApplicationStatus.notAFit,
    ),
);
