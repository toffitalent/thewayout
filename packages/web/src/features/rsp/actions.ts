import { ClientType, RspClientStatus } from '@two/shared';
import { API } from '@app/api';
import type { State } from '@app/store';
import { createApiThunk } from '@app/utils';

export const createRsp = createApiThunk('rsp/create', API.rsp.create);

export const updateRsp = createApiThunk('rsp/update', API.rsp.update);

export const createMember = createApiThunk('rsp/createMember', API.rsp.createMember);

export const inviteCaseManager = createApiThunk('rsp/inviteCaseManager', API.rsp.inviteCaseManager);

export const removeInvitation = createApiThunk('rsp/removeInvitation', API.rsp.removeInvitation);

export const removeCaseManager = createApiThunk('rsp/removeCaseManager', API.rsp.removeCaseManager);

export const listCaseManagers = createApiThunk(
  'rsp/caseManagers',
  ({ rspId }: { rspId: string }, { getState }) => {
    const { ids } = (getState() as State).rsp.caseManagers;
    const lastId = [...ids].pop() as string;
    return API.rsp.listCaseManagers(rspId, {
      before: lastId,
    });
  },
);

export const listInvitations = createApiThunk(
  'rsp/invitations',
  ({ rspId }: { rspId: string }, { getState }) => {
    const { ids } = (getState() as State).rsp.invitations;
    const lastId = [...ids].pop() as string;
    return API.rsp.listInvitations(rspId, {
      before: lastId,
    });
  },
);

export const listCaseManagersNames = createApiThunk(
  'rsp/listCaseManagersNames',
  API.rsp.listCaseManagersNames,
);

export const retrieveRspAccount = createApiThunk('rsp/retrieveRspAccount', API.rsp.retrieveAccount);

export const updateCaseManager = createApiThunk('rsp/updateCaseManager', API.rsp.updateCaseManager);

export const listClients = createApiThunk(
  'rsp/clients',
  ({ rspId }: { rspId: string }, { getState }) => {
    const { ids } = (getState() as State).rsp.clients;
    const lastId = [...ids].pop() as string;
    return API.rsp.listClients(rspId, {
      before: lastId,
    });
  },
);

export const listStatisticsClients = createApiThunk(
  'rsp/statisticsClients',
  (
    {
      rspId,
      // eslint-disable-next-line @typescript-eslint/no-unused-vars
      forceReload,
      ...clientFilters
    }: Partial<ClientType> & { rspId: string; forceReload?: boolean },
    { getState },
  ) => {
    const { ids } = (getState() as State).rsp.statisticsClients;
    const lastId = [...ids].pop() as string;
    return API.rsp.listClients(rspId, {
      ...clientFilters,
      before: lastId,
    });
  },
);

export const retrieveClient = createApiThunk('rsp/retrieveClient', API.rsp.retrieveClient);

export const declineClient = createApiThunk(
  'rsp/declineClient',
  ({ rspId, clientId }: { rspId: string; clientId: string }) =>
    API.rsp.changeClientStatus(rspId, clientId, RspClientStatus.closed),
);

export const acceptClient = createApiThunk(
  'rsp/acceptClient',
  ({ rspId, clientId }: { rspId: string; clientId: string }) =>
    API.rsp.changeClientStatus(rspId, clientId, RspClientStatus.active),
);

export const addClientNotes = createApiThunk('rsp/addClientNotes', API.rsp.addClientNotes);

export const assignCaseManager = createApiThunk(
  'rsp/assignCaseManager',
  ({
    rspId,
    clientId,
    caseManagerId,
  }: {
    rspId: string;
    clientId: string;
    caseManagerId: string;
  }) => API.rsp.assignCaseManager(rspId, clientId, caseManagerId),
);

export const getStatistics = createApiThunk('rsp/statistics', API.rsp.getStatistics);
