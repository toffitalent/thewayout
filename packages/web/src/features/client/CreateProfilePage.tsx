import { Box, Toast, Wizard } from '@disruptive-labs/ui';
import { useRouter } from 'next/router';
import { SEO } from '@app/components/SEO';
import { WizardLayout } from '@app/components/WizardLayout';
import { useAppDispatch, useAppSelector, useErrorIndicator } from '@app/hooks';
import { getClientDataFromWizard } from '@app/utils';
import { selectAuthUser } from '../auth';
import { createClient } from './actions';
import styles from './CreateProfilePage.module.scss';
import { steps } from './profileData';
import { WizardData } from './profileDataTypes';

export const CreateProfilePage = () => {
  const user = useAppSelector(selectAuthUser);
  const dispatch = useAppDispatch();
  const showError = useErrorIndicator();
  const router = useRouter();

  const handleSave = (data: WizardData) => {
    dispatch(createClient(getClientDataFromWizard(data, true)))
      .unwrap()
      .catch((e) => {
        showError(e);
      })
      .then(() => {
        Toast.success('Profile added!');
      });
  };

  if (!user) {
    return null;
  }
  if (user.client) {
    router.push('/client/');
    return null;
  }

  const lastStep = steps.pop();
  if (lastStep) {
    lastStep.nextStep = (data) => handleSave(data as WizardData);
    steps.push(lastStep);
  }

  return (
    <>
      <SEO title="Create Profile" description="" />
      <Box className={styles.wizardWrapper}>
        <Wizard steps={steps} layout={WizardLayout} />
      </Box>
    </>
  );
};
