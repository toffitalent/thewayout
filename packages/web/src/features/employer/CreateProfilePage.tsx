import { Box, Wizard, WizardStep } from '@disruptive-labs/ui';
import { useRouter } from 'next/router';
import { useCallback } from 'react';
import { EmployerCreateRequest as WizardData } from '@two/shared';
import { SEO } from '@app/components/SEO';
import { WizardLayout } from '@app/components/WizardLayout';
import { useAppDispatch, useAppSelector, useErrorIndicator } from '@app/hooks';
import { selectAuthUser } from '../auth';
import { createEmployer } from './actions';
import { ConfirmationStep } from './ConfirmationStep';
import styles from './CreatorProfile.module.scss';
import { steps } from './profileData';

export const CreateProfilePage = () => {
  const user = useAppSelector(selectAuthUser);
  const router = useRouter();
  const dispatch = useAppDispatch();
  const showError = useErrorIndicator();

  const handleSubmit = useCallback(
    (data: WizardData) => {
      if (user) {
        dispatch(createEmployer(data))
          .unwrap()
          .then(() => {
            router.push('/employer/create-profile/confirm/');
          })
          .catch((e) => {
            showError(e);
          });
      }
    },
    [dispatch, router, showError, user],
  );

  if (!user) {
    return null;
  }

  const confirmationStep: WizardStep = {
    id: 'confirm',
    title: 'Confirm your data',
    component: <ConfirmationStep onSubmit={handleSubmit} />,
    showBackButton: true,
  };

  return (
    <>
      <SEO title="Create Profile" description="" />
      <Box className={styles.wizardWrapper}>
        <Wizard steps={[...steps, confirmationStep]} layout={WizardLayout} />
      </Box>
    </>
  );
};
