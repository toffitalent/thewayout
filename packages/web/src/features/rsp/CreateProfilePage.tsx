import { Box, Toast, Wizard } from '@disruptive-labs/ui';
import { useRouter } from 'next/router';
import { RspRole, VeteranOrJustice } from '@two/shared';
import { SEO } from '@app/components/SEO';
import { WizardLayout } from '@app/components/WizardLayout';
import { useAppDispatch, useAppSelector, useErrorIndicator } from '@app/hooks';
import { selectAuthUser } from '../auth';
import { createMember, createRsp } from './actions';
import styles from './CreateProfilePage.module.scss';
import {
  memberProfileSteps,
  ownerProfileSteps,
  RspSteps,
  rspSteps,
  WizardRspData,
} from './rspData';

const getRspRequestData = (data: WizardRspData) => {
  const {
    phone,
    position,
    avatar,
    offenseDrugs,
    offenseMotorVehicle,
    offensePropertyDamage,
    offenseSexual,
    offenseTheft,
    offenseViolent,
    offenseWhiteCollar,
    phoneRspContact,
    emailRspContact,
    avatarRsp,
    ...rsp
  } = data;

  delete rsp.avatarUrl;
  delete rsp.avatarRspUrl;

  return {
    ...rsp,
    avatar: avatarRsp,
    phone: phoneRspContact,
    email: emailRspContact,
    offenses: [
      ...(offenseDrugs || []),
      ...(offensePropertyDamage || []),
      ...(offenseSexual || []),
      ...(offenseTheft || []),
      ...(offenseViolent || []),
      ...(offenseWhiteCollar || []),
      ...(offenseMotorVehicle || []),
    ],
    owner: { phone, position, avatar },
  };
};

export const CreateProfilePage = () => {
  const user = useAppSelector(selectAuthUser);
  const router = useRouter();
  const showError = useErrorIndicator({ ResourceConflict: 'Organization name already exist' });
  const dispatch = useAppDispatch();

  const handleSaveRsp = (data: WizardRspData) => {
    dispatch(createRsp(getRspRequestData(data)))
      .unwrap()
      .catch((e) => {
        showError(e);
      })
      .then(() => {
        Toast.success('Organization added!');
      });
  };

  const handleSaveMember = (data: WizardRspData) => {
    dispatch(
      createMember({
        ...data,
        userId: user?.id as string,
        rspId: user?.rspAccount?.rspId as string,
      }),
    )
      .unwrap()
      .catch((e) => {
        showError(e);
      })
      .then(() => {
        Toast.success('Profile added!');
      });
  };

  if (!user) {
    return null;
  }
  if (user.rspAccount?.isProfileFilled) {
    router.push(user.rspAccount.role === RspRole.owner ? '/ssp/case-managers' : '/ssp/clients');
    return null;
  }

  const isOwner = user.rspAccount?.role === RspRole.owner;

  let steps = memberProfileSteps;
  if (isOwner) {
    steps = [...memberProfileSteps, ...ownerProfileSteps, ...rspSteps];
  }

  if (isOwner) {
    const indexOfContact = steps.findIndex((el) => el.id === RspSteps.contact);
    steps[indexOfContact].nextStep = (data) => {
      if ((data as WizardRspData).veteranOrJustice.includes(VeteranOrJustice.justiceImpacted)) {
        return RspSteps.justiceStatus;
      }
      handleSaveRsp(data);
      return RspSteps.contact;
    };
  }

  const lastStep = steps.pop();
  if (lastStep) {
    lastStep.nextStep = (data) => (isOwner ? handleSaveRsp(data) : handleSaveMember(data));
    steps.push(lastStep);
  }

  return (
    <>
      <SEO title="" description="" />
      <Box className={styles.wizardWrapper}>
        <Wizard
          steps={steps}
          layout={WizardLayout}
          initialData={user.phone ? ({ phone: user.phone } as any) : undefined}
        />
      </Box>
    </>
  );
};
