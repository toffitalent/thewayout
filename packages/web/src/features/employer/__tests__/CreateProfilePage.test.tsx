import userEvent from '@testing-library/user-event';
import { useRouter } from 'next/router';
import { Industry, NumberOfEmployers, State, YearsInBusiness } from '@two/shared';
import { act, fireEvent, fixtures, mockThunks, render, waitFor } from '@test';
import * as actions from '../actions';
import { CreateProfilePage } from '../CreateProfilePage';

const { user } = fixtures;

describe('CreateProfilePage', () => {
  let push: jest.Mock;

  beforeEach(() => {
    push = jest.fn();

    (useRouter as jest.Mock).mockImplementation(() => ({
      isReady: true,
      query: {},
      push,
      beforePopState: jest.fn(),
    }));
  });

  test('renders correctly', async () => {
    const { container, findByRole } = await waitFor(() => render(<CreateProfilePage />));
    await waitFor(async () =>
      expect(await findByRole('textbox', { name: 'Name' })).toBeInTheDocument(),
    );
    expect(container).toMatchSnapshot();
  });

  test('handles submit', async () => {
    const [createEmployer] = mockThunks(actions, ['createEmployer']);
    const { findByRole, findAllByRole, getByRole, getByText, dispatch } = await waitFor(() =>
      render(<CreateProfilePage />),
    );

    // Employer Name step
    await userEvent.type(await findByRole('textbox', { name: 'Name' }), 'Test Company', {
      delay: 1,
    });
    await act(() => fireEvent.click(getByRole('button', { name: 'Save & Next' })));
    expect(await findByRole('heading', { name: 'Industry' })).toBeInTheDocument();

    // Industry step
    await userEvent.selectOptions(await findByRole('combobox'), 'Accounting');
    await act(() => fireEvent.click(getByRole('button', { name: 'Save & Next' })));
    expect(await findByRole('heading', { name: 'Company Description' })).toBeInTheDocument();

    // Description step
    await userEvent.type(await findByRole('textbox', { name: 'Description' }), 'Test Description', {
      delay: 1,
    });
    await act(() => fireEvent.click(getByRole('button', { name: 'Save & Next' })));
    expect(await findByRole('heading', { name: 'Company Details' })).toBeInTheDocument();

    // Company details step
    await userEvent.selectOptions((await findAllByRole('combobox'))[0], '0-3 years');
    await userEvent.selectOptions((await findAllByRole('combobox'))[1], '1-50');
    await act(() => fireEvent.click(getByRole('button', { name: 'Save & Next' })));
    expect(await findByRole('heading', { name: 'Company Location' })).toBeInTheDocument();

    // Company location step
    await userEvent.type(await findByRole('textbox', { name: 'Address' }), 'Test Address', {
      delay: 1,
    });
    await userEvent.type(await findByRole('textbox', { name: 'City' }), 'Test City', {
      delay: 1,
    });
    await userEvent.selectOptions(await findByRole('combobox'), 'Alabama');
    await userEvent.type(await findByRole('textbox', { name: 'ZIP Code' }), 'ZIP', {
      delay: 1,
    });
    await act(() => fireEvent.click(getByRole('button', { name: 'Save & Next' })));
    expect(await findByRole('heading', { name: 'Confirm your data' })).toBeInTheDocument();

    // Confirmation
    expect(getByText('Test Company')).toBeInTheDocument();
    expect(getByText('Accounting')).toBeInTheDocument();
    expect(getByText('Test Description')).toBeInTheDocument();
    expect(getByText('0-3')).toBeInTheDocument();
    expect(getByText('1-50')).toBeInTheDocument();
    expect(getByText('Test Address')).toBeInTheDocument();
    expect(getByText('Test City, AL ZIP')).toBeInTheDocument();
    // Submit
    await act(() => fireEvent.click(getByRole('button', { name: 'Confirm' })));
    await waitFor(async () => {
      expect(createEmployer).toBeCalledWith({
        name: 'Test Company',
        industry: Industry.accounting,
        description: 'Test Description',
        yearsInBusiness: YearsInBusiness['0-3'],
        numberOfEmployees: NumberOfEmployers['1-50'],
        address: 'Test Address',
        city: 'Test City',
        state: State.AL,
        postalCode: 'ZIP',
      });
    });
    expect(dispatch).toBeCalled();
    expect(push).toBeCalledWith('/employer/create-profile/confirm/');
  });
});
