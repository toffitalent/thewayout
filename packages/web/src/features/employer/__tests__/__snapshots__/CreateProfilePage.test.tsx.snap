// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CreateProfilePage renders correctly 1`] = `
<div>
  <div
    class="wizardWrapper"
  >
    <section
      class="layout"
    >
      <div
        class="container"
      >
        <header
          class="header"
        >
          <button
            aria-label="Previous"
            class="button sm outline c-grey disabled iconButton sm round back backHidden"
            disabled=""
            type="button"
          >
            <svg
              aria-hidden="true"
              class="icon"
              focusable="false"
              stroke-width="2"
              viewBox="0 0 24 24"
            >
              <path
                d="M0 0h24v24H0z"
                fill="none"
                stroke="none"
              />
              <polyline
                points="15 6 9 12 15 18"
              />
            </svg>
          </button>
          <div
            aria-valuemax="100"
            aria-valuemin="0"
            aria-valuenow="0"
            class="progress md rounded trackColor progress"
            role="progressbar"
          >
            <div
              class="track bgcolor-primary"
              style="transform: translateX(-100%);"
            />
          </div>
          <h2
            class="text title"
          >
            Employer Name
          </h2>
        </header>
        <form
          class="form"
        >
          <div
            class="control"
            role="group"
          >
            <label
              class="label"
              for=":r0:"
            >
              Name
            </label>
            <div
              class="root primary md fluid md"
            >
              <input
                class="input"
                id=":r0:"
                name="name"
                placeholder="E.g. ABC Company"
                type="text"
                value=""
              />
            </div>
          </div>
          <button
            class="button md solid c-primary disabled fluid submit"
            disabled=""
            type="submit"
          >
            Save & Next
          </button>
        </form>
      </div>
    </section>
  </div>
</div>
`;
